Framework/API/src/IFunction.cpp

```diff
@@ -901,7 +901,7 @@ void IFunction::setMatrixWorkspace(
               if (fitParam.getConstraintPenaltyFactor().compare("")) {
                 try {
                   double penalty =
-                      atof(fitParam.getConstraintPenaltyFactor().c_str());
+                      std::stof(fitParam.getConstraintPenaltyFactor().c_str());
```

Comment:
1. `std::stof` returns a float, but  `penalty` is a double.
2. remove `.c_str()`